### PR TITLE
Rewrite jsonCompiler using StandardCompiler

### PR DESCRIPTION
**Pull request by axic**
Preferably this should be merged after the 0.4.11 release.

Depends on #2114.

### CODE REVIEW

**Comment by axic** [comment]
For some reason this is able to crash the fuzzer:
```
Testing soljson via the fuzzer...
scripts/../test/cmdlineTests.sh: line 80: 12897 Segmentation fault      (core dumped) "$REPO_ROOT"/build/test/solfuzzer --quiet < "$f"
```

**Comment by axic** [comment]
The crash was fixed by #2263.

**Comment by axic** [comment]
@chriseth how do we feel about this one? Probably more tests wouldn't hurt before merging this.

File: solc/jsonCompiler.cpp [hunk 1]
```diff
@@ -131,121 +115,106 @@ Json::Value estimateGas(CompilerStack const& _compiler, string const& _contract)
...
+			output["sources"][sourceName]["AST"] = ret["sources"][sourceName]["legacyAST"];
 		}
+	}
 
-		try
-		{
-			// Do not taint the internal error list
-			ErrorList formalErrors;
-			ErrorReporter errorReporter(formalErrors);
-			if (compiler.prepareFormalAnalysis(&errorReporter))
```
Comment (axic) [inline solc/jsonCompiler.cpp:216]: Note: this removes Why3 support.

**Review by chriseth — APPROVED** [review]
Are there any tests for this?

**Comment by axic** [comment]
There is the `test/libsolidity/JSONCompiler` suite. This change didn't affect the tests.

**Comment by chriseth** [comment]
Compiled this with emscripten and ran the solc-js tests - they are green. Merging and checking in the next days if remix works with it.

**Pull request merged**